### ISSUE TITLE
Allow for CollapsingHeader to be forced open

### ISSUE THREAD

**Issue by Baezon** [emilk/egui#874]
I have collapsing headers which can be indirectly triggered by other UI interactions, so I would like some way to directly set their collapsed state. Perhaps as a function on it, like `default_open()`?

```rust
CollapsingHeader::new(name)
    .forced_open(tree_value == self.tree_view_selection)
    .show(ui, |ui| {
        ...
    });```

**Comment by plasticartsshow** [comment]
It really does sound like `default_open` already solves this. For instance, the CollapsingHeader in the posted snippet will always be open as long as `tree_value == self.tree_view_selection` evaluates to true, so with a bit of handling, this can even override user clicks (like having a node remain forced open)

**Comment by Baezon** [comment]
`default_open` only sets the *initial* state, I'm not entirely sure how this works under the hood, but I'm pretty sure it's based on the `id`, as long as that stays the same, `default_open` will not mess with its state again; it only affects its initial, default open-ness.

I also don't want to determine 'open-ness' entirely, only open where it would otherwise be closed given a particular condition is true. Overriding it entirely would make it impossible for the user to interact with. Just making sure that was clear.

**Comment by plasticartsshow** [comment]
Ah ok, sorry.

**Comment by emilk** [comment]
Seems like a reasonable addition, and a simple PR!

**Comment by Baezon** [comment]
Implemented in #1538, thanks!
